web/quiz_views.py

```
@@ -360,87 +611,148 @@ def _process_quiz_taking(request, quiz):
-            # Calculate percentage score
-            percentage = (score / total_points * 100) if total_points > 0 else 0
+            if ai_auto_correction and correction_status == "in_progress":
+                correction_status = "completed"
+                raw = ai_quiz_corrector(AI_data)
+                ai_correction_results = {}
+
+                # if it comes back as text, parse it
+                if isinstance(raw, str):
+                    ai_correction_results = json.loads(raw)
+                else:
+                    ai_correction_results = raw
```

Ruff (0.8.2)

Use ternary operator `ai_correction_results = json.loads(raw) if isinstance(raw, str) else raw` instead of `if`-`else`-block

Replace `if`-`else`-block with `ai_correction_results = json.loads(raw) if isinstance(raw, str) else raw`

(SIM108)